File: tests/benchmarks/loop/loop.js

// Note that javascript in Node/chrome (V8) is JIT compiled
// which makes more faster than other bytecode interpreted
// VM language listed here.

var start = process.hrtime();

var i=0, list=[];
for (; i < 10000000; i++) list.push(i);

var sum = 0;
for (i=0; i < list.length; i++) sum += list[i];
console.log(sum);

var end = process.hrtime(start);
var secs = (end[0] + end[1] / 1e9).toFixed(6) + 's';
console.log('elapsed:', secs);

File: tests/benchmarks/loop/loop.lua
local start = os.clock()

local list = {}
for i = 1, 10000000 do
	list[i] = i - 1
end

local sum = 0

for i = 1, #list do
	sum = sum + list[i]
end

print(sum)

local seconds = os.clock() - start
print('elapsed: ' .. seconds .. 's')
